apps/internal/base/base.go

```diff
@@ -313,7 +327,7 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
-			return ar, err
+			return ar.ApplyAuthnScheme(&authParams)
 		}
```

Comment:
From @bgavrilMS 

We need to make sure that the token cache schema is the same across MSALs. The simplest is to serialize the cache to a plaintext file and observe it's structure. Note that each access token has a "cache key" and a "cache payload".

For POP tokens, we don't want the `kid` to be part of the access token cache key. This is to prevent the token cache becoming too large, as keys can rotate frequently. 

The changes needed are: 

- add the "schema name" (i.e. "pop") in the key. You'll have to add it to `IAuthenticationScheme` as well. 
- add the `kid` to the payload payload
- ensure `token_type` is in the payload as well

```json
"AccessToken": {
    "-login.windows.net-accesstoken_with_authscheme-16dab2ba-145d-4b1b-8569-bf4b9aed4dc8-72f988bf-86f1-41af-91ab-2d7cd011db47-https://vault.azure.net/.default-pop": {
      "environment": "login.windows.net",
      "client_id": "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8",
      "secret": "actual_token",
      "credential_type": "AccessToken",
      "realm": "72f988bf-86f1-41af-91ab-2d7cd011db47",
      "target": "https://vault.azure.net/.default",
      "cached_at": "1688149697",
      "expires_on": "1688236096",
      "extended_expires_on": "1688236096",
      "kid": "TJtUe-3Wa8X6cOxKzmeTxjHDO1-gXBKtazWh0xx3SA0",    # new!
      "token_type": "pop",   # not Bearer!
      "refresh_on": "1688192896",
      "ext_expires_on": "1688236096"
    }
```

Reply:
In the current design in MSALs, access token bound to key1 overwrites access token bound to key2. Is this a problem?

Reply:
I think so. There can be races around pop key rotation. Ex. around a time pop key rotates from pop_old to pop_new, and  Aq*Token(pop_old) and Aq*Token(pop_new) come in together, MSAL wouldn't know which token to cache and may end up re-writing cache entry w/ pop_old based entry.

Reply:
These cache keys are for MSAL internal use only, not for distributed caches. The race would resolve itself on a single machine pretty quickly. 

A cache key for the distributed cache should include the kid, I agree. MSAL Go doesn't have explicit support for this scenario, yet.

Reply:
@bgavrilMS still working to incorporate your feedback and assess impact on clients due to possible races.

Reply:
@bgavrilMS pushed changes inline with your asks -
 - replaced  `keyid` from accesstoken cache key, replaced with `token_type`
 - added `keyid` to `token_type` to payload. 
 - added `AccessTokenType()` to AuthenticationScheme interface (inline with similar function name in msal .net)

Reply:
Reviewed and left a comment. The access token cache key should not include "-Bearer" (for backwards compat). Only chnage the key for pop tokens. See other comment for more details.

Reply:
Done.